lib/matplotlib/axis.py

```diff
@@ -1517,16 +1517,13 @@ def set_units(self, u):
         """
         if u == self.units:
             return
-        if self is self.axes.xaxis:
-            shared = [
-                ax.xaxis
-                for ax in self.axes.get_shared_x_axes().get_siblings(self.axes)
-            ]
-        elif self is self.axes.yaxis:
-            shared = [
-                ax.yaxis
-                for ax in self.axes.get_shared_y_axes().get_siblings(self.axes)
-            ]
+        for name, axis in self.axes._get_axis_map().items():
+            if self is axis:
+                shared = [
+                    ax._get_axis_map()[name]
```

Comment:
Maybe should do `getattr(ax, f'{name}axis')` directly?

Reply:
Sure, I guess that's a bit more efficient too...